nonlinear_solid_mechanics/include/parameter_handling.h

```diff
@@ -0,0 +1,417 @@
+        output_interval = prm.get_integer("Output interval");
+      }
+      prm.leave_subsection();
+    }
+
+
+    struct NewmarkParameters
+    {
+      double beta;
+      double gamma;
+
+      static void
+      declare_parameters(ParameterHandler &prm);
+
+      void
+      parse_parameters(ParameterHandler &prm);
+    };
+
+    void
+    NewmarkParameters::declare_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("Newmark parameters");
+      {
+        prm.declare_entry("beta", "0.25", Patterns::Double(0, 0.5), "beta");
+
+        prm.declare_entry("gamma", "0.5", Patterns::Double(0, 1), "gamma");
+      }
+      prm.leave_subsection();
+    }
+
+    void
+    NewmarkParameters::parse_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("Newmark parameters");
+      {
+        beta  = prm.get_double("beta");
+        gamma = prm.get_double("gamma");
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    struct PreciceConfiguration
+    {
+      std::string  scenario;
+      bool         enable_precice;
+      std::string  config_file;
+      std::string  participant;
+      std::string  mesh_name;
+      std::string  read_data_name;
+      std::string  write_data_name;
+      unsigned int interface_mesh_id;
+
+      static void
+      declare_parameters(ParameterHandler &prm);
+
+      void
+      parse_parameters(ParameterHandler &prm);
+    };
+
+
+    void
+    PreciceConfiguration::declare_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("precice configuration");
+      {
+        prm.declare_entry("Scenario",
+                          "FSI3",
+                          Patterns::Selection("FSI3|PF"),
+                          "Cases: FSI3 or PF for perpendicular flap");
+        prm.declare_entry(
+          "Enable precice",
+          "true",
+          Patterns::Bool(),
+          "Whether preCICE is used for coupling to another solver");
+        prm.declare_entry("precice config-file",
+                          "precice-config.xml",
+                          Patterns::Anything(),
+                          "Name of the precice configuration file");
+        prm.declare_entry(
+          "Participant",
+          "dealiisolver",
+          Patterns::Anything(),
+          "Name of the participant in the precice-config.xml file");
+        prm.declare_entry(
+          "Mesh name",
+          "dealii-mesh-nodes",
+          Patterns::Anything(),
+          "Name of the coupling mesh in the precice-config.xml file");
+        prm.declare_entry(
+          "Read data name",
+          "received-data",
+          Patterns::Anything(),
+          "Name of the read data in the precice-config.xml file");
+        prm.declare_entry(
+          "Write data name",
+          "calculated-data",
+          Patterns::Anything(),
+          "Name of the write data in the precice-config.xml file");
+        prm.declare_entry(
+          "Interface mesh ID",
```

Comment:
Dangerous as it is easy to mix up with the mesh id of preCICE. In other codes this is called "patch id". If "mesh ID" is the standard dealii name we could call it "deallii mesh id" or similar

Reply:
I agree. Conceptually, we should maybe remove it from the parameter file. The IDs need to be set inside the `make_grid` function i.e. in the source code. Therefore, the here mentioned interface mesh id is just a variable and could be set to any other (but not the same as already used for other boundaries) as well. What do you think?

Reply:
I think I need to make this clear/document this just in the source code